### PR TITLE
Use truncated case notation

### PR DESCRIPTION
**Pull request by nickdrozd**

### CODE REVIEW

File: src/Core/Core.idr [hunk 1]
```diff
@@ -449,10 +449,10 @@ coreFail e = MkCore (pure (Left e))
...
 wrapError : (Error -> Error) -> Core a -> Core a
 wrapError fe (MkCore prog)
-    = MkCore (prog >>=
-                 (\x => pure $ case x of
-                             Left err => Left (fe err)
-                             Right val => Right val))
+    = MkCore $ prog >>=
+                    pure . \case
+                      Left err => Left $ fe err
+                      Right val => Right val
```
Comment (gallais) [inline src/Core/Core.idr:455]: As suggested earlier, this is a good opportunity to use `mapFst` instead.

File: src/Core/Unify.idr [hunk 2]
```diff
@@ -1437,18 +1437,18 @@ retryGuess mode smode (hid, (loc, hname))
...
-                                       pure False -- progress not made yet!
-                                _ => do logTermNF "unify.retry" 5 ("Search failed at " ++ show rig ++ " for " ++ show hname)
-                                                  [] (type def)
-                                        case smode of
-                                             LastChance =>
-                                                 throw !(normaliseErr err)
-                                             _ => pure False) -- Postpone again
+                     \case
+                       DeterminingArg _ n i _ _ =>
+                         let msg = "Failed (det " ++ show hname ++ " " ++ show n ++ ")" in
```
Comment (gallais) [inline src/Core/Unify.idr:1442]: I wouldn't be surprised if you need to add `Lazy String` annotations here to keep the
same semantics. Same for the other branch.

File: src/Parser/Rule/Package.idr [hunk 3]
```diff
@@ -20,117 +20,119 @@ EmptyRule = Grammar () Token False
 export
 equals : Rule ()
 equals = terminal "Expected equals"
```
Comment (gallais) [inline src/Parser/Rule/Package.idr:22]: For all of these, I am tempted to suggest the addition of a `Eq Token` implementation
in `Parser.Lexer.Package` and then turn them all into `guard (TOK ==)`.

File: src/Core/Core.idr (hunk 1 above)
Comment (nickdrozd) [reply src/Core/Core.idr]: added

File: src/Core/Unify.idr (hunk 2 above)
Comment (nickdrozd) [reply src/Core/Unify.idr]: I put the messages back in the function calls.

File: src/Parser/Rule/Package.idr (hunk 3 above)
Comment (nickdrozd) [reply src/Parser/Rule/Package.idr:22]: Certainly the existing arrangement feels a little less elegant than it could be.

File: src/Core/Core.idr [hunk 4]
```diff
@@ -448,7 +448,7 @@ coreFail e = MkCore (pure (Left e))
 
 export
 wrapError : (Error -> Error) -> Core a -> Core a
-wrapError fe (MkCore prog) = MkCore $ prog <&> mapFst fe
+wrapError fe (MkCore prog) = MkCore $ prog >>= pure . mapFst fe
```
Comment (fabianhjr) [inline src/Core/Core.idr:451]: This is fancier though, `<&>` for `>>= pure .`.

Comment (gallais) [reply src/Core/Core.idr:451]: Fancier in the sense it's not shipped by the previous version.

Comment (fabianhjr) [reply src/Core/Core.idr:451]: Oh right

Comment (buzden) [reply src/Core/Core.idr:451]: If `<&>` is unavailable, then use `<$>`, it's definitely available. I.e.
```idris
wrapError fe (MkCore prog) = MkCore $ mapFst fe <$> prog
```
IMO, still better than `>>= pure .`

**Pull request merged**